src/features/indexing/api/index-audio.ts

```diff
@@ -0,0 +1,157 @@
+  const incomingTrackData = await Promise.allSettled(
+    audioFiles
+      .filter(({ id }) => !unmodifiedTracks.has(id))
+      .map(async ({ id, uri, duration, modificationTime }) => {
+        try {
+          const { metadata } = await getAudioMetadata(uri, wantedTags);
+          if (!metadata.artist) throw new Error("Track has no artist.");
+          if (!metadata.name) throw new Error("Track has no name.");
+          return {
+            ...{ id, uri, duration, modificationTime, ...metadata },
+            ...{ artist: metadata.artist!, name: metadata.name! },
+          };
+        } catch (err) {
+          // Propagate error, changing its message to be the track id.
+          if (!(err instanceof Error))
+            console.log(`[Track ${id}] Rejected for unknown reasons.`);
+          else console.log(`[Track ${id}] ${err.message}`);
+          throw new Error(id);
+        }
+      }),
+  );
+  console.log(
+    `Got metadata of ${incomingTrackData.length} tracks in ${stopwatch.lapTime()}.`,
+  );
+
+  // Add rejected tracks to `InvalidTracks` table.
+  await Promise.allSettled(
+    incomingTrackData.filter(isRejected).map(async ({ reason }) => {
+      const trackId = reason.message as string;
+      // Delete existing rejected track as we failed to modify it.
+      if (modifiedTracks.has(trackId)) await deleteTrack(trackId);
+
+      const { id, uri, modificationTime } = audioFiles.find(
+        ({ id }) => id === trackId,
+      )!;
+      await db
+        .insert(invalidTracks)
+        .values({ id, uri, modificationTime })
+        .onConflictDoUpdate({
+          target: invalidTracks.id,
+          set: { modificationTime },
+        });
+    }),
+  );
+
+  // Get all the valid track metadata.
+  const validTrackData = incomingTrackData
+    .filter(isFulfilled)
+    .map(({ value }) => value);
+
+  // Add new artists to database.
+  await Promise.allSettled(
+    [...new Set(validTrackData.map(({ artist }) => artist))].map((name) =>
+      db.insert(artists).values({ name }).onConflictDoNothing(),
+    ),
+  );
+
+  // Add new albums to database (albums may have the same name, but different artists).
+  const albumInfoMap: Record<
+    string,
+    { album: string; artist: string; year: number | null }
+  > = {};
+  const newAlbums = new Set(
+    validTrackData
+      .filter(({ album }) => !!album)
+      .map(({ album, artist, year }) => {
+        // An artist can releases multiple albums with the same name (ie: Weezer).
+        const key = `${album} ${artist} ${year}`;
+        albumInfoMap[key] = { album: album!, artist, year: year ?? null };
+        return key;
+      }),
+  );
+  const albumIdMap: Record<string, string> = {};
+  await Promise.allSettled(
+    [...newAlbums].map(async (albumKey) => {
+      const { album, artist, year } = albumInfoMap[albumKey]!;
+      let exists = allAlbums.find(
+        (a) =>
+          a.name === album && a.artistName === artist && a.releaseYear === year,
```

Comment:
Previously had:
```ts
(a) => a.name === album && a.artistName === artist,
```
Now we have:
```ts
(a) => a.name === album && a.artistName === artist && a.releaseYear === year,
```

When we fixed how we determine if an album is "unique" due to artists potentially having multiple albums with the same name, we forgot to update this check when finding if the album with these specifications already exists.